File: Intersection of Two Arrays II.cpp
class Solution {
public:
    vector<int> intersect(vector<int>& nums1, vector<int>& nums2) {
        map<int, int> counter;
        for (int num: nums1) {
            counter[num]++;
        }

        vector<int> ans;
        for (int num: nums2) {
            if (counter[num]) {
                counter[num]--;
                ans.push_back(num);
            }
        }
        return ans;
    }
};

File: Intersection of Two Arrays II.py
class Solution:
    def intersect(self, nums1: List[int], nums2: List[int]) -> List[int]:
        counter = Counter(nums1)
        ans = []
        for num in nums2:
            if counter[num] > 0:
                ans.append(num)
                counter[num] -= 1
        return ans

